src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -230,6 +230,33 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             }
         }
 
+        /// <summary>
+        /// Gets an &lt;input&gt; elements "type" attribute value for the given <paramref name="modelExplorer"/>.
+        /// </summary>
+        /// <param name="modelExplorer">The <see cref="ModelExplorer"/> to inspect.</param>
+        /// <param name="inputTypeHint">When this method returns, contains a type hint associated with the given
+        /// <paramref name="modelExplorer"/>.</param>
```

Comment:
I wonder whether we should instead expose a `protected string GetInputType(ModelExplorer modelExplorer)` method. The `inputTypeHint` value is meaningless unless you're reading the heuristics in `GetInputTypeHints()` or how `GetFormat()` uses the string. Neither "type hint" nor "associated with the given ..." helps much.

If @marcusien or others find this value useful, it's closer to `... contains the string, often the name of a <see cref="ModelMetadata.ModelType"/> base class, used to determine this method's return value.`